lnst/Recipes/ENRT/ConfigMixins/OffloadSubConfigMixin.py

```diff
@@ -28,6 +28,9 @@ def offload_nics(self):
         raise NotImplementedError("Subclass must implement this property")
 
     def generate_sub_configurations(self, config):
+        if not self.params.offload_combinations:
+            yield config
```

Comment:
this isn't correct, doing this you skip the collaborative inheritance design of other mixins that may be later in the MRO list.

I think this should be part of the loop a couple of lines lower:

```python
for parent_config in super().generate_sub_configurations(config):
    if not self.params.offload_combinations: yield parent_config
```

or i think you could leave it here but do:
```python
if not self.params.offload_combinations:
    yield from super().generate_sub_configurations(config)
```

Reply:
Right, I missed this. Thanks for catching this.